ruby/github_repo_handler.rb

```diff
@@ -17,34 +18,32 @@ def get_account_name
     STDIN.gets.chomp
   end
 
+# **********         Main clone repository logic.       **********
+  def clone_repositories
+    return unless connection? && user_exists?
+
+    calulate_number_of_requests_to_send().times do |index|
+      response_page_logic(index+1)  # starting at one because page=0 and page=1 are identical.
```

Comment:
This isn't your fault, but I'm wondering now why the last page wouldn't have failed, as the last index plus 1 wouldn't exist. I suppose the API returns nothing but this is a case I would like to check by hand. In any case we can just skip the 0 index by throwing `next if index == 0` before `response_page_logic`.

Reply:
Lets imagine we learn the organization/user has 629 repos.
from this equation 
```ruby 
(github_user_info_request()['public_repos']/30.to_f).ceil 
```
```ruby
result = (615/30.to_f).ceil 
# result is 21
```
Both Github and Gitlab start the numbering at 1.

https://gitlab.com/api/v4/users/michaelgdimmitt/projects?page=
https://gitlab.com/api/v4/users/michaelgdimmitt/projects?page=0
https://gitlab.com/api/v4/users/michaelgdimmitt/projects?page=1

Are all equivalent. Maybe the websites are treating empty or 0 to be a value of 1.

Since the counting starts at 0+1.
The value of 0+1+0 will be the 1st page.
The value 0+1+20 will be the 21st page.

We do not count all the way to 21 
our index starts at 0 and ends at 20.

.times() works like a traditional for loop.
```ruby
20.times do |index|
  puts index
end
# output: 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19
```

Reply:
What if we could val+1 with remove the comment replaced with next unless function_determines_valid?
```ruby 
next if index == 0 before response_page_logic
```
```ruby
next unless counter_starts_at_one(index)

def counter_starts_at_one(index) 
  if index > 0
end
```

Reply:
looks like we settled on:
```ruby
calulate_number_of_requests_to_send().times do |index|
      response_page_logic(human_count(index))
end
```
def human_count(index)
  index+1
end
```